### PR TITLE
[Edit profile] Add settings for email preferences and their periodicity of getting them

### PR DESCRIPTION
**Pull request by kryzanivska-nastya**
[#7467](https://github.com/ita-social-projects/GreenCity/issues/7467), [#7544](https://github.com/ita-social-projects/GreenCity/issues/7544), [#7491](https://github.com/ita-social-projects/GreenCity/issues/7491), [#7614](https://github.com/ita-social-projects/GreenCity/issues/7614)
##
Summary of work 🔥

- [x] Extended the form with email preferences, added checkboxes, updated the form submission logic so the user can choose which types of notification he/she wants to receive by email, specifically there are types like: System(by default are on), Comments, Invites, Likes, Places.
- [x]  Implemented the periodicity of sending them. (immediately, twice_a_day, daily, weekly, monthly, never)

https://github.com/user-attachments/assets/b2e11cac-e24f-4dc6-b17c-3e0a48f4ac49

### CODE REVIEW

**Review by kovalsofiia1 — APPROVED** [review]

File: src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts [hunk 1]
```diff
@@ -17,7 +20,61 @@ export class EditProfileFormBuilder {
       showLocation: [false],
       showEcoPlace: [false],
       showShoppingList: [false],
-      socialNetworks: ['']
+      socialNetworks: [''],
+      emailPreferences: this.createEmailPreferencesGroup(null)
+    });
+  }
+
+  private createEmailPreferencesGroup(preferences: any[] | null): FormGroup {
```
Comment (Lokankara) [inline src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts]: preferences has type  NotificationPreference[]
any  type unsafety and makes the code hard to understand and maintain

File: src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts [hunk 2]
```diff
@@ -17,7 +20,61 @@ export class EditProfileFormBuilder {
...
+
+  mapPreferenceToFormControl(preference: string): string | null {
+    return this.emailPrefs.find((pref) => pref.toUpperCase() === preference) || null;
+  }
+
+  capitalizeFirstLetter(string: string): string {
+    return string.charAt(0).toUpperCase() + string.slice(1);
+  }
+
+  getSelectedEmailPreferences(form: FormGroup): any[] {
```
Comment (Lokankara) [inline src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts]: why any[] type, but not NotificationPreference[]

File: src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts [hunk 3]
```diff
@@ -9,6 +9,9 @@ export class EditProfileFormBuilder {
     private builder: FormBuilder,
     private langService: LanguageService
   ) {}
+
+  emailPrefs = ['system', 'likes', 'comments', 'invites', 'places'];
```
Comment (Lokankara) [inline src/app/main/component/user/components/profile/edit-profile/edit-profile-form-builder.ts]: Please rename it to `emailPreferences` to reflect the full naming convention.

File: src/app/main/component/user/components/profile/edit-profile/edit-profile.component.ts [hunk 4]
```diff
@@ -59,6 +59,23 @@ export class EditProfileComponent extends FormBaseComponent implements OnInit, O
...
 
+  emailPreferencesList = [
+    { controlName: 'system', translationKey: 'system', periodicityControl: 'periodicitySystem' },
+    { controlName: 'likes', translationKey: 'likes', periodicityControl: 'periodicityLikes' },
+    { controlName: 'comments', translationKey: 'comments', periodicityControl: 'periodicityComments' },
+    { controlName: 'invites', translationKey: 'invites', periodicityControl: 'periodicityInvites' },
+    { controlName: 'places', translationKey: 'places', periodicityControl: 'periodicityPlaces' }
+  ];
+
+  periodicityOptions = [
```
Comment (Lokankara) [inline src/app/main/component/user/components/profile/edit-profile/edit-profile.component.ts]: Using an enum or constants  is a great way to improve maintainability, type safety, and clarity in your code
you can put into some file like as `models/edit-profile-const.ts`

File: src/app/main/component/user/components/profile/edit-profile/edit-profile.component.ts [hunk 5]
```diff
@@ -148,24 +167,47 @@ export class EditProfileComponent extends FormBaseComponent implements OnInit, O
         this.socialNetworksToServer = data.socialNetworks.map((sn) => sn.url);
         this.coordinates.latitude = data.userLocationDto?.latitude || null;
         this.coordinates.longitude = data.userLocationDto?.longitude || null;
+        this.populateEmailPreferences(data.notificationPreferences);
         this.getFormInitialValues(data);
       });
   }
 
+  populateEmailPreferences(preferences: any[]): void {
```
Comment (Lokankara) [inline src/app/main/component/user/components/profile/edit-profile/edit-profile.component.ts]: populateEmailPreferences(preferences: NotificationPreference[]): void {

**Review by Lokankara — APPROVED** [review]

**Review by hnativlyubomyr — APPROVED** [review]

**Pull request merged**